FreeRTOS/Test/CMock/smp/multiple_priorities_no_timeslice_tickless_idle/covg_multiple_priorities_no_timeslice_tickless_idle_utest.c

```diff
@@ -0,0 +1,153 @@
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ *
+ * https://www.FreeRTOS.org
+ * https://github.com/FreeRTOS
+ *
+ */
+/*! @file single_priority_no_timeslice_covg_utest */
+
+/* C runtime includes. */
+#include <stdlib.h>
+#include <stdbool.h>
+#include <stdint.h>
+#include <string.h>
+
+/* Task includes */
+#include "FreeRTOS.h"
+#include "FreeRTOSConfig.h"
+#include "event_groups.h"
+#include "queue.h"
+
+/* Test includes. */
+#include "unity.h"
+#include "unity_memory.h"
+#include "../global_vars.h"
+#include "../smp_utest_common.h"
+
+/* Mock includes. */
+#include "list.h"
+#include "mock_timers.h"
+#include "mock_fake_assert.h"
+#include "mock_fake_port.h"
+
+#define taskTASK_YIELDING       ( TaskRunning_t ) ( -2 )
+
+
+/* ===========================  EXTERN VARIABLES  =========================== */
+extern volatile UBaseType_t uxCurrentNumberOfTasks;
+extern volatile UBaseType_t uxDeletedTasksWaitingCleanUp;
+extern volatile UBaseType_t uxSchedulerSuspended;
+extern volatile TCB_t *  pxCurrentTCBs[ configNUMBER_OF_CORES ];
+extern volatile BaseType_t xSchedulerRunning;
+extern volatile TickType_t xTickCount;
+extern List_t xSuspendedTaskList;
+extern List_t xPendingReadyList;
+extern volatile UBaseType_t uxTopReadyPriority;
+extern volatile BaseType_t xYieldPendings[ configNUMBER_OF_CORES ];
+extern List_t pxReadyTasksLists[ configMAX_PRIORITIES ];
+
+
+/* ==============================  Global VARIABLES ============================== */
+TaskHandle_t xTaskHandles[configNUMBER_OF_CORES] = { NULL };
+
+/* ============================  Unity Fixtures  ============================ */
+/*! called before each testcase */
+void setUp( void )
+{
+    commonSetUp();
+}
+
+/*! called after each testcase */
+void tearDown( void )
+{
+    commonTearDown();
+}
+
+/*! called at the beginning of the whole suite */
+void suiteSetUp()
+{
+}
+
+/*! called at the end of the whole suite */
+int suiteTearDown( int numFailures )
+{
+    return numFailures;
+}
+
+/* ===========================  EXTERN FUNCTIONS  =========================== */
+extern void vTaskEnterCritical(void);
+
+/* ==============================  Helper functions for Test Cases  ============================== */
+
+
+/* ==============================  Test Cases  ============================== */
+
+/**
+ * @brief xTaskGenericNotify - function to notify a task.
+ *
+ * <b>Coverage</b>
+ * @code{c}
+ *          case eSetValueWithoutOverwrite:
+ *
+ *                    if( ucOriginalNotifyState != taskNOTIFICATION_RECEIVED )
```

Comment:
This line is already covered by other test cases. This line is also not covered by this test cases ( confirmed by gdb ).
Can you describe more about the purpose of this test cases?

Reply:
I like to have explicit cases for each branch in case other tests which implicitly cover the branch change.

Reply:
Just adding more data here. I ran the full test suite for this PR with this test enabled, and then again with this test suite disabled and got identical results for this function. I doesn't appear that this test is providing new coverage not found elsewhere.

Also, it doesn't look like this configuration (multiple priorities / no time slice / tickless idle) is a dependency of the test. There is a code block with #if configUSE_TICKLESS_IDLE but it is not dependent on the other SMP options. In that case a test should be added to the tests found in /tasks (but again this looks like its already covered in other tests anyways)